### ISSUE TITLE
Minimum required structure to compile?

### ISSUE THREAD

**Issue by jsumners** [swiftlang/swift-package-manager#8195]
What is the minimum required structure to successfully issue `swift test`? If I have the following setup, when I issue that command I get: "error: 'json': target 'JsonExperimentTests' has overlapping sources:"

`./Package.swift`:
```swift
// swift-tools-version:6.0

import PackageDescription

let package = Package(
  name: "JsonExperiment",

  products: [
    .library(name: "JsonExperiment", targets: ["JsonExperiment"])
  ],

  targets: [
    .target(name: "JsonExperiment", dependencies: []),
    .testTarget(name: "JsonExperimentTests", dependencies: ["JsonExperiment"])
  ]
)
```

`./Sources/company.swift`:
```swift
import Foundation

struct Company : Decodable {
  let name: String
  let catchPhrase: String
  let moto: String
}
```

`./Sources/company_test.swift`:
```swift
import Testing
import Foundation

@Test
func decodeCompany() {
  let input = """
    {
      "name": "Romaguera-Crona",
      "catchPhrase": "Multi-layered client-server neural-net",
      "bs": "harness real-time e-markets"
    }
  """

  let decoded = try JSONDecoder().decode(Company.self, from: input)
}
```

**Comment by jsumners** [comment]
I think I'm getting close to it with https://github.com/jsumners/swift-experiments/tree/ebeb073deddda038183975caeea9c586eae1d633/json, but now I get an error about the `Company` type not being in scope.

**Comment by jsumners** [comment]
Managed to trial and error into a working setup -- https://github.com/jsumners/swift-experiments/tree/57c28883c76aeed46151163c2d7dd68ffd461724/json

It would be very helpful if this were documented in a single place.

**Comment by plemarquand** [comment]
@AnthonyLatsis is correct `swift package init` is the fastest way to generate the minimal required structure of a package, with tests. There is also some more [documentation available](https://github.com/swiftlang/swift-package-manager/blob/main/Documentation/Usage.md), though I think it could be improved to call out that tests are expected to live in the `Tests` folder.

It sounds like you've got something working now, so I'll close this issue out. If you have specific improvements or suggestions please open a new issue.

**Comment by AnthonyLatsis** [comment]
> but now I get an error about the `Company` type not being in scope

Yeah, `Company` is not a module, so you cannot import it like that. You can either import the parent module like you already have, or use an import specifier like this: `import struct Foo.Company`. The module qualifier is always required. And we are not doing a great job at diagnosing this, but just `import Foo` is not enough as you see because `Company` is `internal` (the default access level).

**Comment by jsumners** [comment]
I think what ended up making things work is `@testable import Foo`. Is that doing something to make `Foo`'s symbols available to the external testing package?

**Comment by AnthonyLatsis** [comment]
Yes, [`@testable`](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/attributes#testable) gives access to internal stuff, but no more than that.

**Comment by jsumners** [comment]
Thank you.
